phac-nml/irida-next
# Add puid to samples and projects

## What does this PR do and why?
This PR adds the PUID to the samples tables (in both projects and groups), as well as a badge on the Project and Sample views containing its PUID.

## Screenshots or screen recordings
Project Samples View:
![image](https://github.com/phac-nml/irida-next/assets/82407232/2801bb65-4d6d-4c26-8e05-65cd80a043c7)

Project Samples View with sorting:
![image](https://github.com/phac-nml/irida-next/assets/82407232/4a40bff3-f949-4e6b-bd49-d5faeeb34783)

Project Samples View with sorting and filtering:
![image](https://github.com/phac-nml/irida-next/assets/82407232/5f21060c-bba0-4c1d-ba4c-0430abcb9e5d)

Group Samples View:
![image](https://github.com/phac-nml/irida-next/assets/82407232/43bc76b6-a279-4830-903c-97b46de8de37)

Group Samples View with sorting:
![image](https://github.com/phac-nml/irida-next/assets/82407232/a745a8a5-6c3a-4b52-ab0e-f483e7323c0a)

Group Samples View with sorting and filtering:
![image](https://github.com/phac-nml/irida-next/assets/82407232/6a1983d0-052f-4586-9187-262551e02b55)

Sample with new puid badge and breadcrumb:
![image](https://github.com/phac-nml/irida-next/assets/82407232/98aaf1c1-1ad2-474d-b5ae-2bc752dc29d5)

Project with new puid badge:
![image](https://github.com/phac-nml/irida-next/assets/82407232/325c3132-0bbe-4440-bb0f-2ac3af2e0b50)

## How to set up and validate locally
1. Navigate to the Projects Samples and Groups Samples views and ensure the PUID for samples are in the table.
2. Ensure you can sort and filter by PUID.
3. Navigate to a sample and project, and ensure you can view the PUID by their respective names.
4. On a sample view, ensure the breadcrumb now contains the accurate PUID, and clicking the PUID breadcrumb links you to the same sample.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
